### PR TITLE
feat(twap): twap review order - pt1

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

This PR is based on and supersedes https://github.com/cowprotocol/cowswap/pull/2593
Implements part of https://github.com/cowprotocol/cowswap/issues/2577

Basic TWAP review details

image

- Price
- Slippage
- Limit price
- Minimum received

# To Test

1. On a Gnosis Safe with the handler already updated, go to advanced
2. Fill in the form
3. Click on review
* You should see the fields above properly filled and calculated

# Notes

1. Styling definitely not ready
2. Not yet in the structure proposed on https://github.com/cowprotocol/cowswap/issues/2577, will be the next PR
5. Remaining parts will come in subsequent PRs

### CODE REVIEW

File: src/modules/twap/pure/ConfirmDetailsItem/index.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+import { RowFixed } from 'legacy/components/Row'
+import { MouseoverTooltipContent } from 'legacy/components/Tooltip'
+
+import { TextWrapper } from 'modules/swap/pure/Row/styled'
+import { StyledInfoIcon } from 'modules/swap/pure/styled'
+
+import { Wrapper, Row, Content } from './styled'
+
+type Props = {
```
Comment (anxolin) [inline src/modules/twap/pure/ConfirmDetailsItem/index.tsx:13]: why not `ConfirmDetailsItemProps` and exporting? (just curious)

File: src/modules/twap/pure/ConfirmDetailsItem/index.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import { ReactNode } from 'react'
+
+import { CornerDownRight } from 'react-feather'
+
+import { RowFixed } from 'legacy/components/Row'
+import { MouseoverTooltipContent } from 'legacy/components/Tooltip'
+
+import { TextWrapper } from 'modules/swap/pure/Row/styled'
+import { StyledInfoIcon } from 'modules/swap/pure/styled'
```
Comment (anxolin) [inline src/modules/twap/pure/ConfirmDetailsItem/index.tsx:9]: should these be exported from the index if needed outside? Or these are too generic and should be moved to a even more common module?

File: src/modules/twap/pure/ConfirmDetailsItem/index.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+            <Content>{children}</Content>
+          </Row>
+        </>
+      ) : (
+        children
+      )}
+    </Wrapper>
+  )
+}
```
Comment (anxolin) [inline src/modules/twap/pure/ConfirmDetailsItem/index.tsx:48]: almost feels like a very generic component, is this something only for twap?

**Review by anxolin — APPROVED** [review]

File: src/modules/twap/pure/ConfirmDetailsItem/index.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  return (
+    <Wrapper>
+      {withArrow && <CornerDownRight size={16} />}
+
+      {label ? (
+        <>
+          <Row>
+            <RowFixed>
+              {label && <TextWrapper>{label}</TextWrapper>}
+              {tooltip && (
```
Comment (shoom3301) [inline src/modules/twap/pure/ConfirmDetailsItem/index.tsx:33]: `InfoIcon` provides the same, can we use it here?

File: src/modules/twap/containers/TwapFormWidget/index.tsx [hunk 5]
```diff
@@ -49,8 +46,8 @@ export function TwapFormWidget() {
 
   // Reset output amount when num of parts or input amount are changed
   useEffect(() => {
-    updateRawState({ outputCurrencyAmount: null })
-  }, [updateRawState, numberOfPartsValue, rawInputCurrencyAmount])
+    updateState?.({ outputCurrencyAmount: null })
```
Comment (shoom3301) [inline src/modules/twap/containers/TwapFormWidget/index.tsx:49]: Thank you! Good change

File: src/modules/twap/containers/TwapConfirmDetails/index.tsx [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+import { SlippageRow } from './SlippageRow'
+import * as styledEl from './styled'
+
+type Props = {
+  inputCurrencyInfo: CurrencyPreviewInfo
+  outputCurrencyInfo: CurrencyPreviewInfo
+  isInvertedState: [boolean, Dispatch<SetStateAction<boolean>>]
+}
+
+export function TwapConfirmDetails(props: Props) {
```
Comment (shoom3301) [inline src/modules/twap/containers/TwapConfirmDetails/index.tsx:28]: Would be great to make this component `pure`

File: src/common/pure/RateInfo/index.tsx [hunk 7]
```diff
@@ -150,8 +152,9 @@ export function RateInfo({
   }, [currentIsInverted, activeRate])
 
   const fiatAmount = useMemo(() => {
+    if (noFiat) return null
```
Comment (shoom3301) [inline src/common/pure/RateInfo/index.tsx:155]: Better to double-check with Michel, maybe we should be consistent with this

**Review by shoom3301 — APPROVED** [review]
Some minor comments, can me addressed later

File: src/modules/twap/pure/ConfirmDetailsItem/index.tsx (hunk 1 above)
Comment (alfetopito) [reply src/modules/twap/pure/ConfirmDetailsItem/index.tsx:13]: This part is coming from original Nenad's PR.
Will rename

File: src/modules/twap/pure/ConfirmDetailsItem/index.tsx (hunk 3 above)
Comment (alfetopito) [reply src/modules/twap/pure/ConfirmDetailsItem/index.tsx:48]: Addressed on https://github.com/cowprotocol/cowswap/pull/2630

**Comment by alfetopito** [comment]
To reduce friction, I'll merge the PR as is and address comments after #2630 and #2631

File: src/common/pure/RateInfo/index.tsx (hunk 7 above)
Comment (alfetopito) [reply src/common/pure/RateInfo/index.tsx:155]: Didn't get this one.
What are you proposing?

File: src/modules/twap/containers/TwapConfirmDetails/index.tsx [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+  const { inputCurrencyInfo, outputCurrencyInfo, isInvertedState } = props
+
+  const rateInfoParams = useRateInfoParams(inputCurrencyInfo.amount, outputCurrencyInfo.amount)
+  const allowedSlippage = useAtomValue(twapOrderSlippage)
+
+  // This is the minimum per part
```
Comment (nenadV91) [inline src/modules/twap/containers/TwapConfirmDetails/index.tsx:34]: Is it per part or for total buy amount? 🤔

**Review by nenadV91 — COMMENTED** [review]
Amazing, thanks a lot Leandro 🙏

**Pull request merged**